lib/error/lib/merr.c

```diff
@@ -22,13 +22,15 @@ extern uint8_t __start_hse_merr;
 extern uint8_t __stop_hse_merr;
 
 merr_t
-merr_pack(int errno_value, const enum hse_err_ctx ctx, const char *file, int line)
+merr_pack(const int errno_value, const int16_t ctx, const char *file, const int line)
 {
     merr_t  err = 0;
     int64_t off;
 
+    static_assert(MERR_CTX_SHIFT == sizeof(ctx) * CHAR_BIT,
+        "merr context size does not match the parameter size");
```

Comment:
Why does the size of ctx have anything to do with the number of bits needed to extract it from merr_t ?

Reply:
yea this is bogus. Will remove.